test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -278,33 +278,43 @@ public void SchemaConnectionShouldntEnlist()
         [Test, IssueLink("https://github.com/npgsql/npgsql/issues/1737")]
         public void Bug1737()
         {
-            var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
+            try
             {
-                Pooling = false,
-                Enlist = true
-            };
+                var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
+                {
+                    Pooling = false,
+                    Enlist = true
+                };
 
-            // Case 1
-            using (var scope = new TransactionScope())
-            {
-                using (var conn = OpenConnection(csb))
-                using (var cmd = new NpgsqlCommand("SELECT 1", conn))
-                    cmd.ExecuteNonQuery();
-                scope.Complete();
-            }
+                // Case 1
+                using (var scope = new TransactionScope())
+                {
+                    using (var conn = OpenConnection(csb))
+                    using (var cmd = new NpgsqlCommand("SELECT 1", conn))
+                        cmd.ExecuteNonQuery();
+                    scope.Complete();
+                }
 
-            // Case 2
-            using (var scope = new TransactionScope())
-            {
-                using (var conn1 = OpenConnection(csb))
-                using (var cmd = new NpgsqlCommand("SELECT 1", conn1))
-                    cmd.ExecuteNonQuery();
+                // Case 2
+                using (var scope = new TransactionScope())
+                {
+                    using (var conn1 = OpenConnection(csb))
+                    using (var cmd = new NpgsqlCommand("SELECT 1", conn1))
+                        cmd.ExecuteNonQuery();
 
-                using (var conn2 = OpenConnection(csb))
-                using (var cmd = new NpgsqlCommand("SELECT 1", conn2))
-                    cmd.ExecuteNonQuery();
+                    using (var conn2 = OpenConnection(csb))
+                    using (var cmd = new NpgsqlCommand("SELECT 1", conn2))
+                        cmd.ExecuteNonQuery();
 
-                scope.Complete();
+                    scope.Complete();
+                }
+            }
+            catch (TransactionAbortedException e) when (e.InnerException is PostgresException { SqlState: "55000"/*object_not_in_prerequisite_state*/ })
```

Comment:
Tests that really are supposed to create a prepared transaction can just live in DistributedTransactionTests - the test suite first checks is max_prepared_transactions > 0, and all tests are skipped if not. Those tests also aren't run, since .NET support for distributed transactions only exists for .NET Framework at the moment.

I took a look and think we have some actual issues with the implementation for unpooled connectors, so I opened #3783. Bottom line, it's probably better to not touch this test in this PR since it will need to be modified anyway.

Reply:
Ok.